server/pubsub/pubsub.go

```diff
@@ -1,56 +0,0 @@
-// Copyright 2023 Woodpecker Authors
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-// Package pubsub implements a publish-subscriber messaging system.
-package pubsub
-
-import (
-	"context"
-	"errors"
-)
-
-// ErrNotFound is returned when the named topic does not exist.
-var ErrNotFound = errors.New("topic not found")
-
-// Message defines a published message.
-type Message struct {
-	// ID identifies this message.
-	ID string `json:"id,omitempty"`
-
-	// Data is the actual data in the entry.
-	Data []byte `json:"data"`
-
-	// Labels represents the key-value pairs the entry is labeled with.
-	Labels map[string]string `json:"labels,omitempty"`
-}
-
-// Receiver receives published messages.
-type Receiver func(Message)
```

Comment:
https://docs.nats.io/nats-concepts/core-nats/pubsub

Reply:
Just add the correct interface and functions when you add the alternative pubsub service. It will probably also need a context and should return an error, but its way easier to do that when we are actually adding it. We don't need to guess about that code now!